mpd/mpd.go

```diff
@@ -1099,6 +1100,21 @@ func (as *AdaptationSet) AddNewAccessibilityElement(scheme AccessibilityElementS
 	return accessibility, nil
 }
 
+// AddNewInbandEventStream - Adds a new InbandEventStream Descriptor to an adaptation set
+// uri - Scheme ID URI for the inband event stream
+// val - value for inband event stream
+func (as *AdaptationSet) AddNewInbandEventStream(uri string, val string) error {
+	if len(uri) <= 0 {
+		return ErrInbandEventStreamSchemeUriEmpty
+	}
+	evt := DescriptorType{
+		SchemeIDURI: Strptr(uri),
+		Value:       Strptr(val),
+	}
+	as.InbandEventStream = append(as.InbandEventStream, evt)
```

Comment:
Should this be adding an EventStreamType to be more spec compliant? Either way you could likely optimise by as you have the same code at both the AdaptationSet and Representation levels and at the very least it looks like that would be a Descriptor. This XML version of the spec could be useful: https://github.com/Dash-Industry-Forum/DASH-IF-Conformance/blob/development/DASH/mpdvalidator/schemas/DASH-MPD.xsd#L82 and it has an InBandEventStream being of EventStream type

Reply:
I agree we should use the `EventStream` type here instead https://github.com/zencoder/go-dash/blob/master/mpd/events.go#L5

It looks like older versions of the spec used to be DescriptorType, but newer versions change it to EventStreamType.

You might need to remove the
```
XMLName     xml.Name `xml:"EventStream"`
```
attribute from it though. You might need to add some `omitempty` to other fields too except `schemeIdUri` since all but that one are optional 

I think we need to keep the same code in both AdapatationSet and Representation levels, as they both extend the `RepresentationBaseType` which includes `InbandEventStream`.

Reply:
I was looking at the schema change from 2014 spec to 2019. There is significant change with many new complex type.  Quoting a few below which I found while going through just 25% of the schema. I am thinking it will be better  if there is a separate user story to adopt 2019 spec. Having one field (InbandEventStream) as per 2019 and rest of the structures as per 2014 will be inconsistent. The functionality is unaffected irrespective of this as the mpd output remains the same w.r.t InbandEventStream. Pls let me know if this can be deferred.  

> In MPDtype
> <xs:element name="ServiceDescription" type="ServiceDescriptionType" minOccurs="0" maxOccurs="unbounded"/>
> <xs:element name="InitializationSet" type="InitializationSetType" minOccurs="0" maxOccurs="unbounded"/>
> <xs:element name="InitializationGroup" type="UIntVWithIDType" minOccurs="0" maxOccurs="unbounded"/>
> <xs:element name="InitializationPresentation" type="UIntVWithIDType" minOccurs="0" maxOccurs="unbounded"/>
> <xs:element name="EssentialProperty" type="DescriptorType" minOccurs="0" maxOccurs="unbounded"/>
> <xs:element name="SupplementalProperty" type="DescriptorType" minOccurs="0" maxOccurs="unbounded"/>
> <xs:element name="UTCTiming" type="DescriptorType" minOccurs="0" maxOccurs="unbounded"/>
> <xs:element name="LeapSecondInformation" type="LeapSecondInformationType" minOccurs="0"/>
> 
> In PeriodType
> 	<xs:element name="SupplementalProperty" type="DescriptorType" minOccurs="0" maxOccurs="unbounded"/>
> 	<xs:element name="EmptyAdaptationSet" type="AdaptationSetType" minOccurs="0" maxOccurs="unbounded"/>
> 	<xs:element name="GroupLabel" type="LabelType" minOccurs="0" maxOccurs="unbounded"/>
> 	<xs:element name="Preselection" type="PreselectionType" minOccurs="0" maxOccurs="unbounded"/>
> 
> In EventStreamType
> 	<xs:attribute ref="xlink:type" fixed="simple"/>
> 	<xs:attribute ref="xlink:show" fixed="embed"/>
> 	<xs:attribute name="presentationTimeOffset" type="xs:unsignedLong" default="0"/>
> 
> In EventType				
> 	<xs:attribute name="contentEncoding" type="ContentEncodingType"/>
> 	<xs:attribute name="messageData" type="xs:string">
> In RepresentationBaseType
> 	<xs:attribute ref="xlink:type" fixed="simple"/>
> 	<xs:attribute ref="xlink:show" fixed="embed"/>			
> 	<xs:attribute name="initializationSetRef" type="UIntVectorType"/>

Reply:
This makes sense, I'm ok with using the 2014 spec and deferring a larger update to be more in line with 2019 spec